### PR TITLE
Add 'exactMatch' option to grid-search API and highlight directive

### LINKED ISSUE #2369: Searching with exact match option
### Description  
Add an option to highlight/search cell whose whole content matches the search criteria (e.g. MS Excel - Match entire cell contents). 

 * igniteui-angular version: 6.2.x
 * browser: N/A

- [x] Development
- [x] Update Grid Search sample
- [x] Update Grid Search topic
- [x] Update README, Spec and CHANGELOG

### PR DESCRIPTION
**Pull request by tachojelev**
Closes #2369.  

Add 'exactMatch' option to grid-search API and highlight directive.

### CODE REVIEW

File: projects/igniteui-angular/src/lib/grid/grid.component.ts [hunk 1]
```diff
@@ -4168,18 +4178,31 @@ export class IgxGridComponent implements OnInit, OnDestroy, AfterContentInit, Af
...
+                                row: rowIndex,
+                                column: j,
+                                page: pageIndex,
+                                index: occurenceIndex++,
+                                groupByRecord: groupByRecord,
+                                item: dataRow
+                            });
+                        }
+                    } else {
+                        while (searchIndex !== -1) {
```
Comment (DDavidkov) [inline projects/igniteui-angular/src/lib/grid/grid.component.ts]: Move the initiation of the searchIndex and occurrenceIndex variables in this else block.

File: projects/igniteui-angular/src/lib/grid/grid.component.ts [hunk 2]
```diff
@@ -4168,18 +4178,31 @@ export class IgxGridComponent implements OnInit, OnDestroy, AfterContentInit, Af
...
-
-                        searchValue = searchValue.substring(searchIndex + searchText.length);
-                        searchIndex = searchValue.indexOf(searchText);
+                    if (exactMatch) {
+                        if (searchValue === searchText) {
+                            this.lastSearchInfo.matchInfoCache.push({
+                                row: rowIndex,
+                                column: j,
+                                page: pageIndex,
+                                index: occurenceIndex++,
```
Comment (DDavidkov) [inline projects/igniteui-angular/src/lib/grid/grid.component.ts]: Hardcode this to 0.

File: projects/igniteui-angular/src/lib/directives/text-highlight/text-highlight.directive.spec.ts [hunk 3]
```diff
@@ -75,6 +75,44 @@ describe('IgxHighlight', () => {
...
+        const fix = TestBed.createComponent(HighlightLoremIpsumComponent);
+        fix.detectChanges();
+
+        const component: HighlightLoremIpsumComponent = fix.debugElement.componentInstance;
+        let count = component.highlightText('Lorem', false, true);
+        fix.detectChanges();
+        let spans = fix.debugElement.nativeElement.querySelectorAll('.' + component.highlightClass);
+        expect(spans.length).toBe(0);
+        expect(count).toBe(0);
+
```
Comment (gedinakova) [inline projects/igniteui-angular/src/lib/directives/text-highlight/text-highlight.directive.spec.ts]: Add a step verifying the behavior when exactMatch is false.

File: projects/igniteui-angular/src/lib/grid/grid.search.spec.ts [hunk 4]
```diff
@@ -163,6 +163,137 @@ describe('IgxGrid - search API', () => {
             expect(spans.length).toBe(0);
         });
 
+        it('findNext and findPrev highlight nothing when there is no exact match, regardless of case sensitivity.', () => {
+            const exactMatch = true;
```
Comment (gedinakova) [inline projects/igniteui-angular/src/lib/grid/grid.search.spec.ts]: What is the reason to define this as a constant?

File: projects/igniteui-angular/src/lib/grid/grid.search.spec.ts [hunk 5]
```diff
@@ -163,6 +163,137 @@ describe('IgxGrid - search API', () => {
...
+            spans = fixNativeElement.querySelectorAll('.' + component.highlightClass);
+            activeSpan = fixNativeElement.querySelector('.' + component.activeClass);
+            expect(activeSpan).toBe(spans[0]);
+            expect(spans.length).toBe(1);
+            expect(count).toBe(1);
+        });
+
+        it('findNext and findPrev highlight only exact matches by respecting case sensitivity', () => {
+            grid.getCellByColumn(5, 'JobTitle').update('director of Dev operations');
+            grid.getCellByColumn(6, 'JobTitle').update('director of HR operations');
```
Comment (gedinakova) [inline projects/igniteui-angular/src/lib/grid/grid.search.spec.ts]: Seems like you need just one of these updates.

File: projects/igniteui-angular/src/lib/grid/grid.search.spec.ts [hunk 6]
```diff
@@ -163,6 +163,137 @@ describe('IgxGrid - search API', () => {
...
+
+            grid.clearFilter('JobTitle');
+            fix.detectChanges();
+
+            activeHighlight = grid.nativeElement.querySelector('.' + component.activeClass);
+            highlights = grid.nativeElement.querySelectorAll('.' + component.highlightClass);
+            expect(highlights.length).toBe(1);
+            expect(activeHighlight).toBe(highlights[0]);
+        });
+
```
Comment (gedinakova) [inline projects/igniteui-angular/src/lib/grid/grid.search.spec.ts]: We should probably add a test with editing as well.

**Pull request merged**